mlflow/utils/uri.py

```diff
@@ -281,3 +356,35 @@ def dbfs_hdfs_uri_to_fuse_path(dbfs_uri):
         )
 
     return _DBFS_FUSE_PREFIX + dbfs_uri[len(_DBFS_HDFS_URI_PREFIX) :]
+
+
+def resolve_uri_if_local(local_uri):
+    """
+    if `local_uri` is passed in as a relative local path, this function
+    resolves it to absolute path relative to current working directory.
+
+    :param local_uri: Relative or absolute path or local file uri
+
+    :return: a fully-formed absolute uri path or an absolute filesystem path
+    """
+    from mlflow.utils.file_utils import local_file_uri_to_path
+
+    if local_uri is not None and is_local_uri(local_uri):
+        scheme = get_uri_scheme(local_uri)
+        cwd = pathlib.Path.cwd()
+        local_path = local_file_uri_to_path(local_uri)
+        if not pathlib.Path(local_path).is_absolute():
+            if scheme == "":
+                return cwd.joinpath(local_path).as_posix()
+            local_uri_split = urllib.parse.urlsplit(local_uri)
+            resolved_absolute_uri = urllib.parse.urlunsplit(
+                (
+                    local_uri_split.scheme,
+                    None,
+                    cwd.joinpath(local_path).as_posix(),
+                    local_uri_split.query,
+                    local_uri_split.fragment,
```

Comment:
Is this for a URI that looks like `file:///foo/bar?p=1#fff`

Reply:
@bali0019 Is this a valid file URI? If so, shall we include the query and fragment in the windows case as well?

Reply:
@bali0019 Can you confirm? (I think the answer is yes)

Reply:
yup, that is correct. it just assembles back the URI that preserves original fragment and query.
To be specific, this is the test that confirms the behavior:
https://github.com/mlflow/mlflow/pull/7670/files#diff-9d5ba8ebadf9e5e465e40ff075a78fe5e61dd32525d3f146b440834ad06eb7d5R1887